receiver/prometheusremotewritereceiver/receiver_test.go

```diff
@@ -219,33 +219,123 @@ func TestTranslateV2(t *testing.T) {
+				// Timeseries 1 gauge metric.
+				m1 := sm1.Metrics().AppendEmpty().SetEmptyGauge()
+				dp1 := m1.DataPoints().AppendEmpty()
+				dp1.Attributes().PutStr("d", "e")
+				dp1.Attributes().PutStr("foo", "bar")
+				// Timeseries 2 gauge metric.
+				m2 := sm1.Metrics().AppendEmpty().SetEmptyGauge()
+				dp2 := m2.DataPoints().AppendEmpty()
+				dp2.Attributes().PutStr("d", "e")
+				dp2.Attributes().PutStr("foo", "bar")
 
+				// Resource 2: for job "foo" with instance "bar"
 				rm2 := expected.ResourceMetrics().AppendEmpty()
-				rmAttributes2 := rm2.Resource().Attributes()
-				rmAttributes2.PutStr("service.name", "foo")
-				rmAttributes2.PutStr("service.instance.id", "bar")
-				mAttributes2 := rm2.ScopeMetrics().AppendEmpty().Metrics().AppendEmpty().SetEmptyGauge().DataPoints().AppendEmpty().Attributes()
-				mAttributes2.PutStr("d", "e")
-				mAttributes2.PutStr("foo", "bar")
+				rm2.Resource().Attributes().PutStr("service.name", "foo")
+				rm2.Resource().Attributes().PutStr("service.instance.id", "bar")
+				sm2 := rm2.ScopeMetrics().AppendEmpty()
+				sm2.Scope().SetName(buildName)
+				sm2.Scope().SetVersion(buildVersion)
+				m3 := sm2.Metrics().AppendEmpty().SetEmptyGauge()
+				dp3 := m3.DataPoints().AppendEmpty()
+				dp3.Attributes().PutStr("d", "e")
+				dp3.Attributes().PutStr("foo", "bar")
 
 				return expected
 			}(),
 			expectedStats: remote.WriteResponseStats{},
 		},
+		{
+			name: "provided otel_scope_name and otel_scope_version",
+			request: &writev2.Request{
+				Symbols: []string{
+					"", "__name__", "metric_with_scope",
+					"otel_scope_name", "custom_scope",
+					"otel_scope_version", "v1.0",
+					"job", "service-y/custom",
+					"instance", "instance-1",
+				},
+				Timeseries: []writev2.TimeSeries{
+					{
+						Metadata:   writev2.Metadata{Type: writev2.Metadata_METRIC_TYPE_GAUGE},
+						LabelsRefs: []uint32{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
+						Samples:    []writev2.Sample{{Value: 10, Timestamp: 100}},
+					},
+				},
+			},
+			expectedMetrics: func() pmetric.Metrics {
+				expected := pmetric.NewMetrics()
+				rm := expected.ResourceMetrics().AppendEmpty()
+				// For job "service-y/custom", resource attributes are set as follows:
+				rm.Resource().Attributes().PutStr("service.namespace", "service-y")
+				rm.Resource().Attributes().PutStr("service.name", "custom")
+				rm.Resource().Attributes().PutStr("service.instance.id", "instance-1")
+				sm := rm.ScopeMetrics().AppendEmpty()
+				// Expect the provided scope values.
+				sm.Scope().SetName("custom_scope")
+				sm.Scope().SetVersion("v1.0")
+				m := sm.Metrics().AppendEmpty().SetEmptyGauge()
+				// The datapoint will have no extra attributes because __name__, otel_scope_name/version,
+				// job and instance are filtered out.
+				_ = m.DataPoints().AppendEmpty()
+				return expected
+			}(),
+			expectedStats: remote.WriteResponseStats{},
+		},
+		{
+			name: "missing otel_scope_name/version falls back to buildName/buildVersion",
```

Comment:
Both `missing otel_scope_name/version falls back to buildName/buildVersion` and `valid request` are testing the same scenario, no? In both, we are not passing otel_scope_name/version and expecting that the current value will be the one that comes from buildName/Version. Maybe we can create a test case where the we assign an value to the settings.BuildInfo.Description/Version and validate here, if the value was properly assigned.